### DISCUSSION TITLE
CF takeoff issues. Hopping when takeoff attempted.

### DISCUSSION THREAD

**Discussion by fkorsah** [bitcraze/discussions#1558] [Q&A] [answered]
Hello!

I am trying to get familiar with Crazyflie 2.1+ so I can use it for my research, but I am having some issues even getting it to take off. 

First, I assembled the Crazyflie propellers with the website's recommended way, but the drone would not leave the ground and air was blowing upwards, not toward the ground. Due to this, I put 47-17R on the front left, and 47-17 on the front right. I also changed the back propellers to match the front, so the same type is diagonal to each other.

This got the drone to leave the ground, but it did not take off properly, it bounced up and down multiple times. I tried controlling it with the cfclient GUI and programming with a Python script but the issue taking off persisted. After testing the first assembled drone a bunch of times, It suddenly started working properly. I am trying to fly multiple drones so I assembled other ones but I am facing the same problem takeoff problem.

I was hoping to get some help with this because I can't figure out why this problem persists. I do not think it is an issue with the propeller orientation since the first drone took off and has my adjusted propeller orientation. The biggest difference between the working drone and the ones that don't is a lighthouse deck installed on it. But I don't it makes a difference in the flight because the lighthouse base station is off.

I was wondering if you have encountered this before and how to fix it. Also, please let me know if I can clarify anything about my problem. 

Working drone: https://github.com/user-attachments/assets/7489c580-738e-46bd-b082-d6ee503f71a7

Take off issue video: https://github.com/user-attachments/assets/9ac41f71-4aa8-479f-85e5-0f72a626d704

**Comment by ToveRumar** [comment]
Hi!

I would start of by checking the motor connections, it looks like they might not be given an equal signal. So just make sure the connectors are properly fastened. 
And yes, just double checking that the props are in the correct placement and properly pressed down on the motor pins, is a good place to start. 

You are correct, the lighthouse deck would not make a difference if the base stations are off. Do you have any other decks mounted on the drone?

Hope this helps in locating your issue!

**Reply by fkorsah** [reply]
Thank you for your response, 

I have double-checked the connections and the wires are in there pretty good. I also made sure the propellers were pressed down on the motor pins. My propeller orientation is also the same as the drone that works. 

For the hopping drones I do not have any decks mounted. 

I was also printing the motor and motor requested logs and compared to the working drone, the PWM seems low. The motor.m1 and motor.m1req, for example, are the same but sometimes the m1req is negative and when this happens the m1 is 0. I thought this might be good to know as well. 

Thank you.

**Comment by gemenerik** [comment, marked as answer]
Can you tell me more about the expansion decks you have attached? It looks like the Crazyflie that uses the Lighthouse system takes off successfully, while the other quad has no decks attached? Autonomous flight without any additional positioning system can be problematic. If you are using the take-off / button commands in the client, this could explain the hopping on the quad without additional decks. If you were using a controller, it is probably something else. One way to easily verify it would be to stick your LH deck on the other Crazyflie, make sure it is calibrated, and try the same script/method of taking off.

**Reply by fkorsah** [reply]
Hello, 

I was able to get the crazyflie to work properly. I needed to have the lighthouse base station on and the drone calibrated for it to work just like you suggested. It seems the onboard sensors are not enough to allow it to fly well without some type of deck providing an estimated position. 

Originally, when I put the LH on the drone that wasn't flying, I couldn't even calibrate it. Later I realized I had to flash new firmware on the crazyflie and the deck connected to it. This allowed me to calibrate the drone with the LH and eventually fly it with my python script. 

Thank you for your help!
